packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -822,15 +834,39 @@ async function formNestedStack(
   serviceName?: string,
   skipEnv?: boolean,
 ) {
-  const initTemplateFilePath = path.join(__dirname, '..', 'resources', rootStackFileName);
-  const nestedStack = JSONUtilities.readJson<Template>(initTemplateFilePath);
+  let rootStack;
+  if (FeatureFlags.getBoolean('overrides.project')) {
+    // CFN transform for Root stack
+    rootStack = await transformRootStack(context);
+
+    // get the {deploymentBucketName , AuthRoleName , UnAuthRole from overridded data}
+
+    const metaToBeUpdated = {
+      DeploymentBucketName: rootStack.Resources.DeploymentBucket.Properties.BucketName,
+      AuthRoleName: rootStack.Resources.AuthRole.Properties.RoleName,
+      UnauthRoleName: rootStack.Resources.UnauthRole.Properties.RoleName,
+    };
+    // sanitize this data if needed
+    for (const key of Object.keys(metaToBeUpdated)) {
+      if (typeof metaToBeUpdated[key] === 'object' && 'Ref' in metaToBeUpdated[key]) {
+        delete metaToBeUpdated[key];
+      }
+    }
+    // update amplify meta with updated root stack Info
```

Comment:
we would need to update team-provider-info.json too?

Reply:
Nice catch